src/python/pants/java/nailgun_client.py

```diff
@@ -97,20 +117,34 @@ class NailgunError(Exception):
 
     DESCRIPTION = 'Problem talking to nailgun server'
 
-    def __init__(self, address, pid, wrapped_exc, traceback):
+    _MSG_FMT = """\
+{description} (address: {address}, remote_pid={pid}, remote_pgrp={pgrp}): {wrapped_exc!r}\
+"""
+
+    def __init__(self, address, pid, pgrp, wrapped_exc, traceback):
       self.address = address
       self.pid = pid
+      self.pgrp = pgrp
       self.wrapped_exc = wrapped_exc
       self.traceback = traceback
-      super(NailgunClient.NailgunError, self).__init__(
-        '{} (address: {}{}): {!r}'
-        .format(
-          self.DESCRIPTION,
-          address,
-          ', remote_pid: {}'.format(pid) if pid is not None else '',
-          self.wrapped_exc
-        )
-      )
+
+      # TODO: these should be ensured to be non-None in NailgunClientSession!
```

Comment:
This removed `expects_pid`, which addressed this TODO. But it probably shouldn't have.

Reply:
That's probably right. I will revert that.

Reply:
Reverted!